plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/persistence/LibraryClasspathContainerSerializer.java

```diff
@@ -0,0 +1,109 @@
+  }
+
+  private LibraryContainerStateLocationProvider stateLocationProvider;
+  
+  public LibraryClasspathContainerSerializer() {
+    this(new DefaultStateLocationProvider());
+  }
+  
+  @VisibleForTesting
+  public LibraryClasspathContainerSerializer(LibraryContainerStateLocationProvider stateLocationProvider) {
+    this.stateLocationProvider = stateLocationProvider;
+  }
+
+  public void saveContainer(IJavaProject javaProject, LibraryClasspathContainer container) throws IOException,
+                                                                                                  CoreException {
+    File stateFile = getContainerStateFile(javaProject, container.getPath(), true);
+    if (stateFile == null) {
+      logger.warning("Container state file cannot be created, save failed");
+      return;
+    }
+    try (PrintWriter outputStream = new PrintWriter(stateFile)) {
+      Gson gson = new GsonBuilder().setPrettyPrinting().create();
+      outputStream.print(gson.toJson(new SerializableLibraryClasspathContainer(container)));
+    }
+  }
+
+  public LibraryClasspathContainer loadContainer(IJavaProject javaProject, IPath containerPath) throws IOException,
+                                                                                                       CoreException {
+    File stateFile = getContainerStateFile(javaProject, containerPath, false);
+    if (stateFile == null || !stateFile.exists()) {
+      return null;
+    }
+    try (FileReader fileReader = new FileReader(stateFile)) {
+      Gson gson = new GsonBuilder().create();
+      SerializableLibraryClasspathContainer fromJson =
+          gson.fromJson(fileReader, SerializableLibraryClasspathContainer.class);
+      return fromJson.toLibraryClasspathContainer();
+    }
+  }
+  
+  private File getContainerStateFile(IJavaProject javaProject, IPath containerPath, boolean create) 
+                                                                                                throws CoreException {
+    IPath containerStateFile = stateLocationProvider.getContainerStateFile(javaProject, containerPath, create);
+    if (containerStateFile != null) {
+      return containerStateFile.toFile();
+    } else {
+      return null;
+    }
+  }
+  
+  private static class DefaultStateLocationProvider implements LibraryContainerStateLocationProvider {
+
+    @Override
+    public IPath getContainerStateFile(IJavaProject javaProject, IPath containerPath, boolean create) 
+                                                                                                throws CoreException {
+      IFolder settingsFolder = javaProject.getProject().getFolder(".settings");
+      IFolder folder = settingsFolder.getFolder(FrameworkUtil.getBundle(getClass()).getSymbolicName());
+      if (!folder.exists() && create) {
+        folder.create(true, true, null);
+      }
+      IFile containerFile = folder.getFile(containerPath.segment(1) + ".container");
```

Comment:
What happens if the folder doesn't exist but `create` is false? Will `folder.getFile()` work as intended?